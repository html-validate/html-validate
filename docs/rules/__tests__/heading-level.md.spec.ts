import HtmlValidate from "../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["incorrect"] = `<h1>Heading 1</h1>
<h3>Subheading</h3>`;
markup["correct"] = `<h1>Heading 1</h1>
<h2>Subheading</h2>`;
markup["min-initial-rank"] = `<nav>
    <h2>Navigation</h2>
</nav>
<h1>Heading 1</h1>`;
markup["sectioning-root"] = `<h1>Heading 1</h1>
<h2>Subheading 2</h2>
<dialog>
    <!-- new sectioning root, heading level can restart at h1 -->
    <h1>Dialog header</h1>
</dialog>
<!-- after dialog the level is restored -->
<h3>Subheading 3</h2>`;

describe("docs/rules/heading-level.md", () => {
	it("inline validation: incorrect", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"heading-level":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"heading-level":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: min-initial-rank", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"heading-level":["error",{"minInitialRank":"h2"}]}});
		const report = htmlvalidate.validateString(markup["min-initial-rank"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: sectioning-root", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"heading-level":"error"}});
		const report = htmlvalidate.validateString(markup["sectioning-root"]);
		expect(report.results).toMatchSnapshot();
	});
});
