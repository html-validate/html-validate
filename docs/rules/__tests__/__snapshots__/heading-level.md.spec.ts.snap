// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs/rules/heading-level.md inline validation: correct 1`] = `[]`;

exports[`docs/rules/heading-level.md inline validation: incorrect 1`] = `
[
  {
    "errorCount": 1,
    "filePath": "inline",
    "messages": [
      {
        "column": 2,
        "context": undefined,
        "line": 2,
        "message": "Heading level can only increase by one, expected <h2> but got <h3>",
        "offset": 20,
        "ruleId": "heading-level",
        "ruleUrl": "https://html-validate.org/rules/heading-level.html",
        "selector": "h3",
        "severity": 2,
        "size": 2,
      },
    ],
    "source": "<h1>Heading 1</h1>
<h3>Subheading</h3>",
    "warningCount": 0,
  },
]
`;

exports[`docs/rules/heading-level.md inline validation: min-initial-rank 1`] = `[]`;

exports[`docs/rules/heading-level.md inline validation: sectioning-root 1`] = `[]`;
